draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+### Location
+
+Location behavior is necessary when not all locations within a data format are valid reference targets, or when object properties that appear to be JRI keywords are only actually JRI keywords in certain locations.  In formats where most locations cause automatic actions (such as applying a schema to an instance), location behavior can also create areas where the automatic actions do not apply.  These reserved locations can safely hold re-usable objects separate from any single use.
+
+The passive nature of location behavior makes it particularly suitable to be defined directly by context specifications, as noted in {{context-loc}}.  However, not all uses of JRI involve a context specification.
+
+JRI reserves the `"$defs"` keyword in order to make no-automatic-action location behavior available within the `"$"`-prefixed namespace.  This keyword also enables safe interoperable bundling by standalone tools as described in {{bundling-interop}}.[^70]
+
+[^70]: It's also possible to bundle documents that do not use `"$defs"` into a document that does, and can therefore be un-bundled by a generic bundling tool.  I'm not sure of the best way to talk about that, as I did not want to get into defining a bundling media type.
+
+#### `"$defs"`
+
+The value of the `"$defs"` property MUST be an object, which MUST have objects as the values of all of its properties.  JRI keywords MUST be allowed in these objects.  Aside from their use in JSON Pointers or similar structural identifiers, the property names within the `"$defs"` object MUST NOT be considered to impose specific semantics to the property values.
+
+### Referencing
+
+JRI reference usage has two components: resolving the reference target and interpreting its semantics.  JRI offers several keywords for determining a reference target.  Once determined, regardless of which keywords are used, the possible semantics are as described in {{ref-semantics}}.
+
+#### `"$ref"`
+
+The value of the `"$ref"` property MUST be a string, and MUST be a valid IRI-reference as defined by {{!RFC3987, Section 2.2}}.  The IRI produced by resolving this IRI-reference against the current base IRI identifies the reference target.
+
+By default, the reference target MAY be assumed to be of whatever media type would otherwise be present inline, and fragments MAY be evaluated accordingly.  Context specifications MAY set other requirements regarding the nature of reference targets but MUST NOT expect these requirements to be interoperable with standalone JRI implementations.
+
+#### `"$extRef"`
+
+To avoid overloading the behavior of `"$ref"` or relying on non-interoperable context-specific behavior, `"$extRef"` (for "extensible reference") allows specifying additional metadata as well as alternative mechanisms for secondary resource identification.[^90]
```

Comment:
@DavidBiesack 🤦  It's supposed to be an object, apparently I spliced out that paragraph, thanks for catching it.

The HTTP headers approach is one possibility.  `$extRef` is intentionally under-specified because I'm not entirely sure how it should work and know that there are some ideas floating around out there.  Assuming there's interest in building on this submission once we're able to evaluate other submissions that come in, I suspect we'll spend substantial time on fleshining out `$extRef`.

But I will fix the part about defining it as an object.

I'm generally vehemently against doing anything based on file extensions, but that's more of a personal preference than a considered recommendation, and also the sort of thing can be debated in issues once we have a baseline proposal merged.

Reply:
I agree - we can't rely on extension - that's why we may need to pass `Accept:`.

Reply:
@handrews the update still does not specify that `"$extRef"` is an object. Add:

```
The value of the `"$extRef"` property MUST be an object.
```
Are there other constraints? i.e. MUST it contain a `"$ref"` string (or perhaps another nested `"$extRef"` ?)